packages/logging_cloudwatch/aws_logging_cloudwatch/test/remote_constraint_provider_test.dart

```diff
@@ -0,0 +1,233 @@
+      when(() => mockFileStorage.load(any()))
+          .thenAnswer((_) async => Future.value(sampleJson));
+      when(() => mockFileStorage.save(any(), any())).thenAnswer((_) async {});
+      mockAWSHttpClient = MockAWSHttpClient((request, _) {
+        return AWSHttpResponse(
+          statusCode: 400,
+          body: utf8.encode('NO RESPONSE'),
+        );
+      });
+      provider = BaseRemoteLoggingConstraintProvider.forTesting(
+        config: const DefaultRemoteConfiguration(
+          refreshInterval: Duration(seconds: 10),
+          endpoint: 'https://example.com',
+          region: 'us-west-2',
+        ),
+        fileStorage: mockFileStorage,
+        awsHttpClient: mockAWSHttpClient,
+      );
+
+      await provider.ready;
+
+      // Verify that _loggingConstraint uses local storage
+      expect(
+        provider.loggingConstraint!.toJson(),
+        equals(json.decode(sampleJson)),
+      );
+    });
+
+    test('updates constraints when endpoint returns updated constraints',
+        () async {
+      const updatedJson = '''
+        {
+          "defaultLogLevel": "debug",
+            "categoryLogLevel": {
+                "API": "debug",
+                "AUTH": "error"
+            },
+            "userLogLevel": {
+                "cognito-sub-xyz-123": {
+                    "defaultLogLevel": "verbose",
+                    "categoryLogLevel": {
+                        "API": "error",
+                        "AUTH": "debug"
+                    }
+                  }
+              }
+        }
+        ''';
+      // Mocking the endpoint to return updated constraints on the second call
+      var callCount = 0;
+      var callCount2 = 0;
+      when(() => mockFileStorage.load(any())).thenAnswer((_) async {
+        callCount2++;
+        if (callCount2 == 1) {
+          return Future.value(sampleJson);
+        } else {
+          return Future.value(updatedJson);
+        }
+      });
+      when(() => mockFileStorage.save(any(), any())).thenAnswer((_) async {});
+      mockAWSHttpClient = MockAWSHttpClient((request, _) {
+        callCount++;
+        if (callCount == 1) {
+          return AWSHttpResponse(
+            statusCode: 200,
+            body: utf8.encode(sampleJson),
+          );
+        } else {
+          return AWSHttpResponse(
+            statusCode: 200,
+            body: utf8.encode(updatedJson),
+          );
+        }
+      });
+      provider = BaseRemoteLoggingConstraintProvider.forTesting(
+        config: const DefaultRemoteConfiguration(
+          refreshInterval: Duration(seconds: 1),
+          endpoint: 'https://example.com',
+          region: 'us-west-2',
+        ),
+        fileStorage: mockFileStorage,
+        awsHttpClient: mockAWSHttpClient,
+      );
+
+      // Wait for the provider to refresh the constraints and make the second call to the endpoint
+      await Future<void>.delayed(const Duration(seconds: 2));
```

Comment:
Question: why to delay? I think the `await provider.ready` should be sufficient.